impress-org/givewp
# Fix: PayPal onboarding with email address that has plus sign

Resolves [GIVE-800]

## Description



The PayPal onboarding query params are not encoded, yet we are handling them as if they were encoded by parsing them.  The problematic query param is the `merchantId` (email address) that could potentially have special character like a `+` sign.  Ideally we would be able to encode that value before parsing it.  

This PR does a really simple `str_replace` on plus signs to convert them to the encoded character.

Note: This has been reported to PayPal and they opened a ticket to investigate on there side.  Their response was to handle the special characters on our end.


## Affects



Paypal onboarding

## Visuals



N/A

## Testing Instructions



- connect to paypal using an email with a plus sign
- make sure that plus sign persists
- make sure donations go through as expected

## Pre-review Checklist






[GIVE-800]: https://stellarwp.atlassian.net/browse/GIVE-800?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
